### PR TITLE
JAMES-4050 Allow `%` and `*` characters in mailbox names

### PR DESCRIPTION
**Pull request by quantranhong1999**

### CODE REVIEW

File: docs/modules/servers/pages/distributed/configure/jvm.adoc [hunk 1]
```diff
@@ -103,3 +103,15 @@ Ex in `jvm.properties`
...
 
+== Relax mailbox name validation
+
+The property `james.relax.mailbox.name.validation` allows to accept `*` and `%` characters in mailbox name.
+
+Optional. Boolean. Default to true.
+
+Ex in `jvm.properties`
+----
+james.relax.mailbox.name.validation=false
```
Comment (chibenwa) [inline docs/modules/servers/pages/distributed/configure/jvm.adoc:114]:
```suggestion
james.relaxed.mailbox.name.validation=false
```

File: docs/modules/servers/pages/distributed/configure/jvm.adoc [hunk 2]
```diff
@@ -103,3 +103,15 @@ Ex in `jvm.properties`
 # james.reactor.inputstream.prefetch=4
 ----
 
+== Relax mailbox name validation
+
+The property `james.relax.mailbox.name.validation` allows to accept `*` and `%` characters in mailbox name.
+
+Optional. Boolean. Default to true.
```
Comment (chibenwa) [inline docs/modules/servers/pages/distributed/configure/jvm.adoc:110]: Default to false. Let's avoid breaking changes?

File: docs/modules/servers/pages/distributed/configure/jvm.adoc [hunk 3]
```diff
@@ -103,3 +103,15 @@ Ex in `jvm.properties`
...
+
+Ex in `jvm.properties`
+----
+james.relax.mailbox.name.validation=false
+----
+To strictly validate `*` and `%` characters in the mailbox name.
```
Comment (chibenwa) [inline docs/modules/servers/pages/distributed/configure/jvm.adoc:116]: Please explain that % and * are ambiguous for the LIST / LSUB commands that interprete those as wildcard thus returning all mailboxes mating the pattern.

File: mailbox/api/src/test/java/org/apache/james/mailbox/model/MailboxPathTest.java [hunk 4]
```diff
@@ -367,17 +367,17 @@ void assertAcceptableShouldNotThrowWhenSharpInTheMiddle() {
...
 
     @Test
-    void assertAcceptableShouldThrowOnWildcard() {
-        assertThatThrownBy(() -> MailboxPath.forUser(USER, "a*b")
+    void assertAcceptableShouldNotThrowOnWildcardByDefault() {
+        assertThatCode(() -> MailboxPath.forUser(USER, "a*b")
             .assertAcceptable('.'))
-            .isInstanceOf(MailboxNameException.class);
+            .doesNotThrowAnyException();
     }
```
Comment (chibenwa) [inline mailbox/api/src/test/java/org/apache/james/mailbox/model/MailboxPathTest.java:381]: Please check the two modes.

**Review by chibenwa — COMMENTED** [review]
Please refer to this in jvm.properties files within `server/apps` path.

File: server/apps/jpa-smtp-app/sample-configuration/jvm.properties [hunk 5]
```diff
@@ -53,4 +53,8 @@ jmx.remote.x.mlet.allow.getMBeansFromURL=false
 openjpa.Multithreaded=true
 
 # Whether James should unzip JWTs. Default to false
-# james.jwt.zip.allow=false
\ No newline at end of file
+# james.jwt.zip.allow=false
+
+# Relax validating `*` and `%` characters in the mailbox name. Defaults to false.
+# Be careful turning on this as `%` and `*` are ambiguous for the LIST / LSUB commands that interpret those as wildcard thus returning all mailboxes matching the pattern.
+#james.relaxed.mailbox.name.validation=true
```
Comment (chibenwa) [inline server/apps/jpa-smtp-app/sample-configuration/jvm.properties:60]: This app have no mailbox this option is thus not needed

File: server/apps/scaling-pulsar-smtp/sample-configuration/jvm.properties [hunk 6]
```diff
@@ -46,3 +46,7 @@
 
 # Whether James should unzip JWTs. Default to false
 # james.jwt.zip.allow=false
+
+# Relax validating `*` and `%` characters in the mailbox name. Defaults to false.
+# Be careful turning on this as `%` and `*` are ambiguous for the LIST / LSUB commands that interpret those as wildcard thus returning all mailboxes matching the pattern.
+#james.relaxed.mailbox.name.validation=true
```
Comment (chibenwa) [inline server/apps/scaling-pulsar-smtp/sample-configuration/jvm.properties:52]: Idem this app is smtp only

File: mpt/impl/imap-mailbox/inmemory/pom.xml [hunk 7]
```diff
@@ -81,7 +81,7 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
                 <configuration>
-                    <reuseForks>true</reuseForks>
+                    <reuseForks>false</reuseForks>
```
Comment (quantranhong1999) [inline mpt/impl/imap-mailbox/inmemory/pom.xml:84]: This was needed to retain isolation for `MailboxPath`' static final INVALID_CHARS across test classes.

Acceptable for memory tests IMO.

Comment (chibenwa) [reply mpt/impl/imap-mailbox/inmemory/pom.xml:84]: Ok. What's the overall degradation?

**Review by Arsnael — APPROVED** [review]

File: mpt/impl/imap-mailbox/inmemory/pom.xml (hunk 7 above)
Comment (quantranhong1999) [reply mpt/impl/imap-mailbox/inmemory/pom.xml:84]: On my local PC, for the MPT memory module. Test run time:
Before: 35s
After: 36s

Neglectable. Likely bootstrap new JVMs is too fast on my PC.

**Review by vttranlina — APPROVED** [review]

**Comment by Arsnael** [comment]
Needs a rebase @quantranhong1999

**Pull request merged**